crates/rattler_conda_types/src/repo_data/topological_sort.rs

```diff
@@ -27,11 +30,16 @@ pub fn sort_topologically<T: AsRef<PackageRecord> + Clone>(packages: Vec<T>) ->
 }
 
 /// Find cycles with DFS
-fn find_all_cycles<T: AsRef<PackageRecord>>(packages: &FxHashMap<String, T>) -> Vec<Vec<String>> {
+fn find_all_cycles<T: AsRef<PackageRecord>>(
+    packages: &HashMap<String, T, RandomState>,
+) -> Vec<Vec<String>> {
     let mut all_cycles = Vec::new();
-    let mut visited = FxHashSet::default();
+    let mut visited = HashSet::default();
+
+    let mut package_names: Vec<_> = packages.keys().collect();
+    package_names.sort();
```

Comment:
Another thing I had to change because of the no longer deterministic ordering of the `HashSet`.

Reply:
I prefer we use `IndexMap` for these types to make sure they are deterministic instead of reordering which is more expensive.